apps/asap-server/src/controllers/calendars.ts

```diff
@@ -30,21 +32,50 @@ export default class Calendars implements CalendarController {
     this.squidexGraphqlClient = squidexGraphqlClient;
   }
 
-  async fetch(options: {
-    take: number;
-    skip: number;
-  }): Promise<ListCalendarResponse> {
-    const { take = 50, skip = 0 } = options;
-    const { total, items: calendars } =
-      await this.calendarSquidexRestClient.fetch({
-        take,
+  async fetch(): Promise<ListCalendarResponse> {
+    const take = 50;
+    const skip = 0;
+
+    const { queryCalendarsContentsWithTotal } =
+      await this.squidexGraphqlClient.request<
+        FetchCalendarsQuery,
+        FetchCalendarsQueryVariables
+      >(FETCH_CALENDARS, {
+        top: take,
         skip,
-        sort: [{ path: 'data.name.iv', order: 'ascending' }],
+        filter: '',
+        order: 'data/name/iv asc',
       });
 
+    if (queryCalendarsContentsWithTotal === null) {
+      logger.warn('queryCalendarsContentsWithTotal returned null');
+      return {
+        total: 0,
+        items: [],
+      };
+    }
+
+    const { items: calendars } = queryCalendarsContentsWithTotal;
+
+    if (calendars === null) {
+      logger.warn('queryCalendarsContentsWithTotal items returned null');
+      return {
+        total: 0,
+        items: [],
+      };
+    }
+
+    const activeCalendars = calendars.filter(
+      (calendar) =>
+        calendar.referencingGroupsContents &&
+        calendar.referencingGroupsContents.findIndex(
```

Comment:
This wouldn't work?
calendar.referencingGroupsContents?.findIndex 
so you could remove "calendar.referencingGroupsContents &&" part

Reply:
actually it wouldn't - the part before `?` would resolve to undefined which is not equal to `-1` and would therefore pass the condition, but the requirements are that a calendar without a group (ie empty array or no reference to groups at all) should be skipped.

Reply:
So you could use sth like calendar.referencingGroupsContents?.findIndex(...) ?? false or so :)